### PR TITLE
Added custom random functions

### PR DESCRIPTION
**Pull request by igas**
Related to: #45 #37

### CODE REVIEW

File: lib/faker.ex [hunk 1]
```diff
@@ -76,14 +76,100 @@ defmodule Faker do
...
+  @doc """
+  Returns a positive integer between 0 and total.
+
+  ## Examples
+
+      iex> random(5) in [0, 1, 2, 3, 4, 5]
+      true
+  """
+  @spec random(integer) :: integer
+  def random(total) do
```
Comment (tobyhinloopen) [inline lib/faker.ex]: While your example does hint that the random function should return an integer between 0 and `total` (inclusive), it is not clear it does. I'd suggest mentioning it explicitly in the doc:

> Returns an integer between 0 and `n`

Additionally, I feel `n` to be a better variable name than `total`, but it doesn't really matter to me

**Comment by igas** [comment]
> Without looking at the implementation of a Faker function that uses randomness, it is not clear which random function will be used in the end.

How about putting each of them in the docs? It's rare case from my opinion, so if someone doing this, I assume that he/she knows what they are doing.

> I feel like all functions using some kind of randomness should all pass a single point - they should all end up to a single random function that is configurable.

Not sure how I feel about this one, on one the hand it's simple, on the other I'm not sure it'll keep all flexibility. Also, I don't think I have an idea how to pass all of them through a single function, probably I can try with first 3, but have no idea what to do with random bytes.

> all 4 functions should be implemented

I haven't tested, but I believe you can set only one of them and the rest will be taken from package `env`.

> allow the user to assign a single module that implements a behaviour that requires the 4 random functions to exist

This sounds like a good alternative, we also can provide default implementations, so only needed will be overridden.

Definetely `random/1` not needed and can be implemented in `random_between/1,2`.

**Comment by hauleth** [comment]
Why not just use `Enum.random/1` and `Enum.take_random/2`? Wouldn't that be much simpler?

**Comment by scohen** [comment]
Hey, while upgrading to elixir 1.5 / erl 20.1, this is the last library that we use that emits warnings. Can we have some movement on this PR?

I'd be glad to pitch in if you can use me.

**Comment by igas** [comment]
@hauleth it will be, but it relies on new OTP/Elixir versions, seeds and less customizable.

**Comment by scohen** [comment]
@igas `Enum.random` and  `Enum.take_random` have been part of Elixir's standard lib since 1.1, which came out in September 2015. Also, with the move from `:random` to `:rand`, you're going to break everyone not running on erlang >= 18, which came out around the same time.

**Comment by igas** [comment]
@scohen fair point about versions, we already support 1.3+ due to dates etc, so we could use it, I guess.

@tobyhinloopen can you please take a look at last commit and give feedback?

**Pull request merged**